10up/action-pr-helper
# Add Milestone to PR

### Description of the Change
Add auto-assigning milestone feature. 

This feature helps to automatically add milestone to Pull Requests. If the PR is connected to an issue with a milestone, the same milestone will be added to the PR. Otherwise, the next milestone from the available milestones will be assigned, sorted using version comparison.

Closes #45

### How to test the Change
1. Create a [workflow file](https://github.com/iamdharmesh/action-pr-validator/pull/46/files) using this action and pointing to this branch. (`uses: 10up/action-pr-automator@enhancement/45`)
2. Create one sample issue, assign a milestone to it, and raise a sample PR that fixes the created issue.
3. Verify the same milestone added to PR.
4. Create one more sample issue without milestone and raise a sample PR that fixes the created issue.
5. Verify that milestone added to PR from the available milestones, sorted using version comparison. (For example, if `1.2.1`, `1.3.0` and `Future Release` milestones are available then `1.2.1` will be added.)

### Changelog Entry
> Added - Auto-milestoning feature.

### Credits
Props @iamdharmesh, @jeffpaul.

### Checklist:
